src/React.js

```diff
@@ -1,40 +1,14 @@
 import React from "react";
 
 function createClass(baseClass) {
-  function bindProperty(instance, prop, value) {
-    switch (prop) {
-      case "state":
-      case "render":
-      case "componentDidMount":
-      case "componentWillUnmount":
-        instance[prop] = value;
-        break;
-
-      case "componentDidCatch":
-      case "componentWillUpdate":
-      case "shouldComponentUpdate":
-      case "getSnapshotBeforeUpdate":
-        instance[prop] = function (a, b) { return value(a)(b)(); };
-        break;
-
-      case "componentDidUpdate":
-        instance[prop] = function (a, b, c) { return value(a)(b)(c)(); };
-        break;
-
-      case "unsafeComponentWillMount":
-        instance["UNSAFE_componentWillMount"] = value;
-        break;
-
-      case "unsafeComponentWillReceiveProps":
-        instance["UNSAFE_componentWillReceiveProps"] = function (a) { return value(a)(); };
-        break;
-
-      case "unsafeComponentWillUpdate":
-        instance["UNSAFE_componentWillUpdate"] = function (a, b) { return value(a)(b)(); };
-        break;
-
-      default:
-        throw new Error("[purescript-react] Not a component property: " + prop);
+  function curry2(f) {
+    return f === undefined ? f : function (a, b) {
+      return f(a)(b)
```

Comment:
I don't think these are correct. They are missing the `()` for invoking the effects (ie, it's not just a curry).

Reply:
I see. Previously, I applied the amount of arguments + 1 as the currying number for `UNSAFE_componentWillReceiveProps` and `UNSAFE_componentWillUpdate`, but not sure why I have missed them for the remaining.

Now the invocation and currying should be fixed. Please take another look.